sdks/typescript/src/opik/client/Client.ts

```diff
@@ -0,0 +1,67 @@
+import { loadConfig, OpikConfig } from "@/config/Config";
+import { OpikApiClient } from "@/rest_api";
+import type { Trace as ITrace, ProjectPublic } from "@/rest_api/api";
+import { SavedTrace, Trace } from "@/tracer/Trace";
+import { v7 as uuid } from "uuid";
+
+interface TraceData extends Omit<ITrace, "startTime"> {
+  startTime?: Date;
+}
+
+export class OpikClient {
+  public apiClient: OpikApiClient;
+  public config: OpikConfig;
+  private existingProjects: Map<string, ProjectPublic> = new Map();
+
+  constructor(explicitConfig?: Partial<OpikConfig>) {
+    this.config = loadConfig(explicitConfig);
+
+    this.apiClient = new OpikApiClient({
+      apiKey: this.config.apiKey,
+      environment: this.config.host,
+      workspaceName: this.config.workspaceName,
+    });
+  }
+  public loadProject = async (projectName: string): Promise<ProjectPublic> => {
+    if (this.existingProjects.has(projectName)) {
+      return this.existingProjects.get(projectName)!;
+    }
+
+    const { data: projectsPage } = await this.apiClient.projects
+      .findProjects({ name: projectName })
+      .asRaw();
+
+    const project = projectsPage.content?.find(
```

Comment:
we have the endpoint to get a project by name `api/v1/private/projects/retrieve`

Reply:
Nice call :)